charmcraft/linters.py

```diff
@@ -264,6 +264,13 @@ def run(self, basedir: pathlib.Path) -> str:
             self.text = f"The metadata.yaml file is missing the following attribute(s): {missing}."
             return self.Result.ERRORS
 
+        if "series" in metadata:
+            self.text = (
+                "The metadata.yaml file contains the deprecated attribute: series."
+                "This attribute will be rejected starting in Juju 4.0."
+            )
```

Comment:
Charmcraft implements a [deprecation message subsystem](https://github.com/canonical/charmcraft/blob/main/charmcraft/deprecations.py) that's not currently in use and likely to go away with the craft-application rebase. Reporting this as a linter warning seems appropriate.